internal/datastore/memdb/readwrite.go

```diff
@@ -35,13 +35,18 @@ func (rwt *memdbReadWriteTx) WriteRelationships(mutations []*core.RelationTupleU
 func (rwt *memdbReadWriteTx) write(tx *memdb.Txn, mutations ...*core.RelationTupleUpdate) error {
 	// Apply the mutations
 	for _, mutation := range mutations {
+		cr, err := rwt.prepareCaveat(tx, mutation)
+		if err != nil {
+			return err
+		}
 		rel := &relationship{
 			mutation.Tuple.ResourceAndRelation.Namespace,
 			mutation.Tuple.ResourceAndRelation.ObjectId,
 			mutation.Tuple.ResourceAndRelation.Relation,
 			mutation.Tuple.Subject.Namespace,
 			mutation.Tuple.Subject.ObjectId,
 			mutation.Tuple.Subject.Relation,
+			cr,
```

Comment:
This will need to store the context information written with the tuple as well; we'll need to add that the the `RelationTupleUpdate`

Reply:
@josephschorr that's already in place?

Reply:
No, we'll need to decide how to store it.